### PR TITLE
dash/fix-dts

### PR DESCRIPTION
**Pull request by bre1470**
Improved TypeScript-support of Dashboards.

### CODE REVIEW

File: ts/Data/Connectors/CSVConnectorOptions.d.ts [hunk 1]
```diff
@@ -33,25 +33,25 @@ export interface CSVConnectorOptions extends DataConnectorOptions {
...
      * Whether to enable polling for live data.
      */
-    enablePolling: boolean;
+    enablePolling?: boolean;
     /**
      * Whether to treat the first row of the data set as series names.
      * @default true
      */
-    firstRowAsNames: boolean;
+    firstRowAsNames?: boolean;
```
Comment (sebastianbochan) [inline ts/Data/Connectors/CSVConnectorOptions.d.ts:54]: What is the reason for replacing all params with optional?

`Partial<>` is not enough then?

Comment (bre1470) [reply ts/Data/Connectors/CSVConnectorOptions.d.ts:54]: It is enough, but non-optional would indicate a default value, which is incorrect.

File: ts/Shared/Types.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/* *
+ *
+ *  (c) 2010-2021 Torstein Honsi
```
Comment (karolkolodziej) [inline ts/Shared/Types.ts:3]: Are you sure?

File: ts/Shared/Types.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+'use strict';
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+
+namespace Types {
```
Comment (karolkolodziej) [inline ts/Shared/Types.ts:19]: I'm not convinced do the naming, how about `SharedTypes`  or `GlobalTypes`?

**Review by karolkolodziej — APPROVED** [review]

File: ts/Shared/Types.ts (hunk 3 above)
Comment (bre1470) [reply ts/Shared/Types.ts:19]: This seems to be redundant as it already is in the `Shared`. `Global` would be not accurate as it is not part of the global namespace.

**Pull request merged**